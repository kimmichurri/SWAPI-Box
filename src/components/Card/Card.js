import React from 'react';
import nautical_star from '../../assets/nautical_star.svg';

export default function Card({ card }) {
    console.log(card)
    return (
        <div className="individual-card">
            <h3 className="card-title">{card.name}
            <button className="favorite-container">
                <img 
                    className="favorite-icon" 
                    src={nautical_star} 
                    alt="black and white nautical star" 
                />
            </button>
            </h3>
            <p>{card.species} </p>
            <p>{card.homeworld}</p>
            <p>{card.population}</p>
            <p>{card.language}</p>
            <p>{card.terrain}</p>
            <p>{card.climate}</p>
            <p>{card.residents}</p>
            <p>{card.model}</p>
            <p>{card.class}</p>
            <p>{card.passengers}</p>
        </div>
    )
}