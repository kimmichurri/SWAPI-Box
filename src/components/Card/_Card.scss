.individual-card {
    background-color: $card-color;
    border-radius: 4px;
    box-shadow: $card-box-shadow 5px 5px 20px;
    color: $white;
    height: 400px;
    margin: 20px;
    width: 350px;
}

.card-title {
    background-color: $dark-blue-active;
    box-shadow: black 1px 0px 2px;
    display: flex;
    font-size: 1.5em;
    justify-content: space-between;
    margin: 0px;
    padding: 20px;
}

.favorite-icon {
    width: 50px;
    &:hover {
        transform: rotate(72deg);
        transition: .75s;
    }
}

.favorite-container {
    background-color: $dark-blue-active;
    border: none;
        &:hover {
            cursor: pointer;
        }
}