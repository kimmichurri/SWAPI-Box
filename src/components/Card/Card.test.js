import React from 'react';
import Card from './Card';
import { shallow } from 'enzyme';

describe('Card', () => {
  let wrapper;
  let mockCard;

  beforeEach(() => {
    mockCard= {
      species: 'Human',
      homeworld: 'Earth',
      population: '100',
      language: 'Spanish',
      terrain: 'dirt',
      climate: 'wet',
      residents: 'me',
      model: 'jeep',
      class: 'suv',
      passengers: '4'
    }
    wrapper = shallow(<Card card={mockCard}/>)
  })

  it('should match the snapshot', () => {
    expect(wrapper.debug()).toMatchSnapshot()
  })
})
