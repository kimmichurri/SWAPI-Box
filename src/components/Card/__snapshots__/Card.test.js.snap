// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match the snapshot 1`] = `
"<div className=\\"individual-card\\">
  <h3 className=\\"card-title\\">
    <button className=\\"favorite-container\\">
      <img className=\\"favorite-icon\\" src=\\"nautical_star.svg\\" alt=\\"black and white nautical star\\" />
    </button>
  </h3>
  <p>
    Human
     
  </p>
  <p>
    Earth
  </p>
  <p>
    100
  </p>
  <p>
    Spanish
  </p>
  <p>
    dirt
  </p>
  <p>
    wet
  </p>
  <p>
    me
  </p>
  <p>
    jeep
  </p>
  <p>
    suv
  </p>
  <p>
    4
  </p>
</div>"
`;
